src/lib.rs

```diff
@@ -64,10 +64,104 @@ struct CmdArgs {
     is_local: bool,
 }
 
+/// Run a SAFE vault to join a network
+#[derive(StructOpt, Debug)]
+#[structopt(name = "safe-nlt-join")]
+struct JoinCmdArgs {
+    /// Verbosity level for this tool
+    #[structopt(short = "v", long, parse(from_occurrences))]
+    verbosity: u8,
+
+    /// Path where to locate safe_vault/safe_vault.exe binary. The SAFE_VAULT_PATH env var can be also used to set the path
+    #[structopt(short = "p", long, env = "SAFE_VAULT_PATH")]
+    vault_path: Option<PathBuf>,
+
+    /// Path where the output directories for all the vaults are written
+    #[structopt(short = "d", long, default_value = "./vaults")]
+    vaults_dir: PathBuf,
+
+    /// Verbosity level for vaults logs (default: INFO)
+    #[structopt(short = "y", long, parse(from_occurrences))]
+    vaults_verbosity: u8,
+
+    /// IP used to launch the vaults with.
+    #[structopt(short = "h", long)]
+    hard_coded_contacts: Option<String>,
```

Comment:
This can be easily done [like this](https://github.com/maidsafe/quic-p2p/blob/14e601daed8b80d3008352883d898fc75f93a5d7/src/config.rs#L27-L34)

StructOpt will do the parsing for us

Reply:
short, and long parameter makes sense but there is no need to parse this since it'll be passed to the vaults as a string.